#!/usr/bin/env python

PACKAGE="miradar_node"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


size_enum = gen.enum([ gen.const("Halt", int_t, 0, "Halt"),
                  gen.const("PPI", int_t, 1, "PPI Mode"),
                  gen.const("Map", int_t, 2, "Map Mode"),
                    ],
                  "An enum to set sensor mode")

gen.add("sensor_mode", int_t, 0, "sensor mode", 0, 0, 2, edit_method=size_enum)
gen.add("max_distance", double_t, 0, "Scan Max Distance", 4, 1, 50)
gen.add("min_distance", double_t, 0, "Scan Min Distance", 0.3, 0, 50)
gen.add("alarm_distance", double_t, 0, "Alarm Distance", 1.0, 0, 50)
gen.add("max_angle", int_t, 0, "Scan Max Angle", 44, 10, 90)

gen.add("max_dB", int_t, 0, "Scan Maximum dB", -30, -73, -10)
gen.add("min_dB", int_t, 0, "Scan Minimum dB", -50, -74, -11)


scan_enum = gen.enum([ gen.const("Threshold", int_t, 0, "Threshold Mode"),
                  gen.const("Max", int_t, 1, "Max Mode")],
                  "An enum to set laserscan retreiving mode")

gen.add("scan_mode", int_t, 0, "Laserscan mode", 0, 0, 1, edit_method=scan_enum)


distance_enum = gen.enum([ gen.const("res_32", int_t, 32, "Resolution 32"),
                  gen.const("res_64", int_t, 64, "Resolution 64"),
                  gen.const("res_128", int_t, 128, "Resolution 128"),
                  gen.const("res_256", int_t, 256, "Resolution 256")
                  ],
                  "An enum to set distance division")

gen.add("distance_div", int_t, 0, "Number of Distance Division", 64, 32, 256, edit_method=distance_enum)
gen.add("angle_div", int_t, 0, "Number of Angle Division", 45, 11, 181)

gen.add("tx_power", int_t, 0, "TX Power", -7, -10, 0)

hpf_enum = gen.enum([
    gen.const("hpf_0db", int_t, 0, "+6dB"),
    gen.const("hpf_12db", int_t, 1, "+12dB"),
    gen.const("hpf_18db", int_t, 2, "+18dB")
],
"An enum to set HPF gain"
)

gen.add("hpf_gain", int_t, 0, "HPF Gain", 1, 0, 2, edit_method=hpf_enum)

pga_enum = gen.enum([
    gen.const("pga_0db", int_t, 0, "0dB"),
    gen.const("pga_6db", int_t, 1, "+6dB"),
    gen.const("pga_12db", int_t, 2, "+12dB"),
    gen.const("pga_18db", int_t, 3, "+18dB"),
    gen.const("pga_24db", int_t, 4, "+24dB")
],
"An enum to set PGA gain"
)

gen.add("pga_gain", int_t, 0, "PGA Gain", 1, 0, 3, edit_method=pga_enum)
gen.add("duration", int_t, 0, "duration", 200, 10, 5000)
gen.add("num_ppi_plot", int_t, 0, "num_ppi_plot", 1, 1, 8)
exit(gen.generate(PACKAGE, "miradar_node", "MiRadar"))
